src/js/angular/ttyg/directives/chat-item-detail.directive.js

```diff
@@ -50,13 +58,63 @@ function ChatItemDetailComponent(toastr, $translate, TTYGContextService, Markdow
                 $scope.onRegenerateQuestion({chatItem: $scope.chatItemDetail});
             };
 
-            $scope.copyAnswerToClipboard = () => {
-                $scope.onCopyAnswerToClipboard({chatItem: $scope.chatItemDetail});
+            /**
+             * Extract the explanation of how the answer was generated.
+             */
+            $scope.explainResponse = () => {
+                if (TTYGContextService.hasExplainResponse($scope.chatItemDetail.answer.id)) {
+                    TTYGContextService.toggleExplainResponse($scope.chatItemDetail.answer.id);
+                } else {
+                    $scope.loadingExplainResponse = true;
+                    TTYGService.explainResponse($scope.chatItemDetail)
+                        .then((explainResponse) => {
+                            $scope.explainResponseModel = explainResponse;
+                            TTYGContextService.addExplainResponseCache(explainResponse);
+                        })
+                        .catch(() => {
+                            toastr.error($translate.instant('ttyg.chat_panel.messages.explain_response_failure'));
+                        })
+                        .finally(() => $scope.loadingExplainResponse = false);
+                }
+            };
+
+            /**
+             * Triggers an asking how the answer was generated.
+             */
+            $scope.onAskHowAnswerWasDerived = () => {
+                $scope.onAskHowDeliveredAnswer({chatItem: $scope.chatItemDetail});
+            };
+
+            /**
+             * Opens <code>query</code> in sparql editor.
+             * @param {string} query
+             */
+            $scope.onOpenInSparqlEditor = (query) => {
+                if ($scope.chatItemDetail.agentId) {
+                    const agent = TTYGContextService.getAgent($scope.chatItemDetail.agentId);
+                    if (agent) {
```

Comment:
What's the point in checking for the agent?

Reply:
There is a possibility that the item may not have an agent if it doesn't have an answer yet. However, when the SPARQL page opens, we will always have the agentId. I will remove this check.

Reply:
Removed